stratum/hal/lib/tdi/tdi_ipsec_manager.h

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2022-2023 Intel Corporation.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef STRATUM_HAL_LIB_TDI_IPSEC_MANAGER_H_
+#define STRATUM_HAL_LIB_TDI_IPSEC_MANAGER_H_
+
+#include <map>
+#include <memory>
+
+#include "absl/base/thread_annotations.h"
+#include "absl/memory/memory.h"
+#include "absl/synchronization/mutex.h"
+#include "absl/time/time.h"
+#include "absl/types/optional.h"
+#include "stratum/glue/integral_types.h"
+#include "stratum/hal/lib/common/gnmi_events.h"
+#include "stratum/hal/lib/common/utils.h"
+#include "stratum/hal/lib/common/writer_interface.h"
+#include "stratum/hal/lib/tdi/tdi_fixed_function_manager.h"
+#include "stratum/hal/lib/tdi/tdi_sde_interface.h"
+#include "stratum/lib/channel/channel.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void ipsec_notification_callback(uint32_t dev_id,
+                                 uint32_t ipsec_sa_spi,
+                                 bool soft_lifetime_expire,
+                                 uint8_t ipsec_sa_protocol,
+                                 char *ipsec_sa_dest_address,
```

Comment:
Can this be a `const char *`?

Reply:
This definition comes from what the TDI layer defined the callback mechanism to look like. The definition needs to match how TDI registers callbacks and changes will need to be performed there (and other components that use the callback mechanism such as bfswitchd before we can make any changes here)